go/host/host.go

```diff
@@ -531,8 +531,7 @@ func (h *host) storeBlockProcessingResult(result *common.BlockSubmissionResponse
 	// only update the host rollup headers if the enclave has found a new rollup head
 	if result.FoundNewHead {
 		// adding a header will update the head if it has a higher height
-		headerWithHashes := common.HeaderWithTxHashes{Header: result.IngestedRollupHeader, TxHashes: result.ProducedRollup.TxHashes}
-		h.db.AddRollupHeader(&headerWithHashes)
+		h.db.AddRollupHeader(result.IngestedRollupHeader, result.ProducedRollup.TxHashes)
```

Comment:
While you're here could you add a todo saying there's a bug here and that the tx hashes are being stored against the header for a different rollup?

There's a task for it but I think it's worth noting in the code to help avoid confusion for readers.